TYPO3.Neos/Resources/Private/TypoScript/Prototypes/Menu.ts2

```diff
@@ -6,7 +6,7 @@ prototype(TYPO3.Neos:Menu) < prototype(TYPO3.TypoScript:Template) {
 	node = ${node}
 	items = ${this.items}
 
-	entryLevel = 1
+	entryLevel = ${this.startingPoint ? 0 : 1}
```

Comment:
Why not using a processor to force the value to 0 when starting point is set:

    entryLevel.@process.respectStartingPoint = ${this.startingPoint ? 0 : value}